### PR TITLE
sushi token resolver

### PR DESCRIPTION
**Pull request by krisbitney**
includes sushiswap and sushibar

### CODE REVIEW

File: packages/core-sdk/resolvers/token-resolvers/resolvers/sushi/src/query/constants.ts [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+  AVALANCHE = 43114,
+  AVALANCHE_TESTNET = 43113,
+  HECO = 128,
+  HECO_TESTNET = 256,
+  HARMONY = 1666600000,
+  HARMONY_TESTNET = 1666700000,
+  OKEX = 66,
+  OKEX_TESTNET = 65,
+  CELO = 42220,
+  // PALM = 11297108109, // throws because it doesn't fit in i32
```
Comment (Niraj-Kamdar) [inline packages/core-sdk/resolvers/token-resolvers/resolvers/sushi/src/query/constants.ts:28]: Let's open an issue to have chainId as BigInt in Ethereum_Network type

File: packages/core-sdk/resolvers/token-resolvers/resolvers/sushi/src/query/functions/getTokenComponents.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    args: [],
+    connection: connection,
+  });
+  if (token1AddressResult.isErr) {
+    throw new Error("Invalid protocol token");
+  }
+  return [token0AddressResult.unwrap(), token1AddressResult.unwrap()];
+}
+
+function getSushiSwapComponents(
```
Comment (Niraj-Kamdar) [inline packages/core-sdk/resolvers/token-resolvers/resolvers/sushi/src/query/functions/getTokenComponents.ts:43]: I am wondering whether we should re-use the uniswap token resolver here?

**Review by Niraj-Kamdar — CHANGES_REQUESTED** [review]

File: packages/core-sdk/resolvers/token-resolvers/resolvers/sushi/src/query/constants.ts (hunk 1 above)
Comment (krisbitney) [reply packages/core-sdk/resolvers/token-resolvers/resolvers/sushi/src/query/constants.ts:28]: see PR here: https://github.com/polywrap/monorepo/pull/747

File: packages/core-sdk/resolvers/token-resolvers/resolvers/sushi/src/query/functions/getTokenComponents.ts (hunk 2 above)
Comment (krisbitney) [reply packages/core-sdk/resolvers/token-resolvers/resolvers/sushi/src/query/functions/getTokenComponents.ts:43]: What is the best way to do that? They are in different packages, so should we add the uniswap token resolver to npm and then add it as a dependency in the sushi token resolver??

File: packages/core-sdk/resolvers/token-resolvers/resolvers/sushi/src/query/functions/getTokenComponents.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+export function getTokenComponents(input: Input_getTokenComponents): Interface_TokenComponent {
+  if (env == null) throw new Error("env is not set");
+  const connection = (env as QueryEnv).connection;
+
+  const token = Token_Query.getToken({
+    address: input.tokenAddress,
+    m_type: Token_TokenType.ERC20,
+  }).unwrap();
+
+  if (token.address.toLowerCase() == XSUSHI_ADDRESS.toLowerCase()) {
```
[the remark this answers is not in the document]
Comment (krisbitney) [reply packages/core-sdk/resolvers/token-resolvers/resolvers/sushi/src/query/functions/getTokenComponents.ts]: done

File: packages/core-sdk/resolvers/token-resolvers/resolvers/sushi/src/query/functions/getTokenComponents.ts (hunk 2 above)
Comment (krisbitney) [reply packages/core-sdk/resolvers/token-resolvers/resolvers/sushi/src/query/functions/getTokenComponents.ts:43]: Sorry, I just realized I misread your original comment. They are similar enough that we could combine them into one token resolver. I think same with linkswap. The main difference is in the contract addresses being queried, and sushibar is a little different.

Comment (Niraj-Kamdar) [reply packages/core-sdk/resolvers/token-resolvers/resolvers/sushi/src/query/functions/getTokenComponents.ts:43]: Exactly, but for now, let's keep it as it is and open an issue to investigate this further.

**Review by Niraj-Kamdar — APPROVED** [review]
LGTM!

**Pull request merged**